src/Checkout/ExpressCheckout/Service/PayPalExpressCheckoutDataService.php

```diff
@@ -104,9 +113,19 @@ protected function getButtonLanguageSetting(): string
         return Settings::ECS_BUTTON_LANGUAGE_ISO;
     }
 
-    private function showPayLater(string $salesChannelId, AvailabilityContext $availabilityContext): bool
+    private function showPayLater(string $salesChannelId, AvailabilityContext $availabilityContext, ?ShopwareSalesChannelEvent $event): bool
     {
-        return $this->systemConfigService->getBool(Settings::ECS_SHOW_PAY_LATER, $salesChannelId)
+        return $event instanceof ProductPageLoadedEvent
+            && $this->systemConfigService->getBool(Settings::ECS_SHOW_PAY_LATER, $salesChannelId)
             && $this->payLaterMethodData->isAvailable($availabilityContext);
```

Comment:
So this will return false, if we are on a `CheckoutCartPage`? But PayLater should be shown then.